processor.go

```diff
@@ -256,7 +305,7 @@ func (p *processor) dispatch(clientIP net.Addr, reqID uuid.UUID, m map[string]*p
 			defer wg.Done()
 
 			var r result
-			r.code, r.body, r.err = p.send(clientIP, reqID, tenant, wrReq)
+			r.code, r.body, r.duration, r.err = p.send(clientIP, reqID, tenant, wrReq)
```

Comment:
maybe we should switch to just return `result` struct here? too many fields already :)